File: Q1/ans.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        string input = "mohammad";
        Dictionary<char, int> counts = new Dictionary<char, int>();

        foreach (char c in input)
        {
            if (counts.ContainsKey(c))
                counts[c]++;
            else
                counts[c] = 1;
        }

        foreach (var pair in counts)
        {
            Console.WriteLine($"{pair.Key}: {pair.Value}");
        }
    }
}

File: Q1/ans.cpp
#include <iostream>
#include <map>
#include <string>
using namespace std;

int main() {
    string input = "mohammad";
    map<char, int> counts;

    for (char c : input) {
        counts[c]++;
    }

    for (auto pair : counts) {
        cout << pair.first << ": " << pair.second << endl;
    }

    return 0;
}
